src/main/java/io/vertx/core/http/impl/ServerConnection.java

```diff
@@ -102,14 +102,17 @@
     this.server = server;
     this.handshaker = handshaker;
     this.metrics = metrics;
-    this.metric = metrics.connected(remoteAddress());
   }
 
   @Override
-  protected Object metric() {
+  protected synchronized Object metric() {
```

Comment:
yes, this is an handback object provided by the metrics SPI that contains data associated with the connection